js/htmldiff.js

```diff
@@ -109,12 +123,19 @@
    * @param {number} start_in_after The index of the first token in the list of after tokens.
    * @param {number} length The number of consecutive matching tokens in this block.
    */
-  function Match(start_in_before, start_in_after, length){
-    this.start_in_before = start_in_before;
-    this.start_in_after = start_in_after;
+  function Match(start_in_before, start_in_after, length, segment){
+    this.segment = segment;
+    this.start_in_before = start_in_before + segment.before_index;
+    this.start_in_after = start_in_after + segment.after_index;
     this.length = length;
-    this.end_in_before = (this.start_in_before + this.length) - 1;
-    this.end_in_after = (this.start_in_after + this.length) - 1;
+    
+    this.segment_start_in_before = start_in_before;
+    this.segment_start_in_after = start_in_after;
+    this.segment_end_in_before = (this.segment_start_in_before + this.length) - 1;
+    this.segment_end_in_after = (this.segment_start_in_after + this.length) - 1;
+
+    this.end_in_before = segment.before_index + this.segment_end_in_before;
+    this.end_in_after = segment.after_index + this.segment_end_in_after;
```

Comment:
These were hard for me to process. Think it'd be clearer as

    this.end_in_before = this.start_in_before + this.length - 1;
    this.end_in_after = this. start_in_after + this.length - 1;